### DISCUSSION TITLE
Using gitoxide to replace apache git-http-backend

### DISCUSSION THREAD

**Discussion by casibbald** [GitoxideLabs/gitoxide#362] [Q&A] [answered]
I am looking at how I can possibly build a basic Git HTTPS service that allows (clone, push) for branches and tags.

Googling around, I came across gitoxide and wondered if the following script alias examples are now replaceable?

```
ScriptAliasMatch \
  "(?x)^/git/(.*/)((HEAD | \
    info/refs | \
    objects/(info/[^/]+ | \
    [0-9a-f]{2}/[0-9a-f]{38} | \
    pack/pack-[0-9a-f]{40}\.(pack|idx)) | \
    git-(upload|receive)-pack))$" \
    "/var/www/html/git/$1/git-core/git-http-backend/$2"
```

The intention is to build a git mirror for an air-gapped environment.

**Comment by Byron** [comment, marked as answer]
Thanks for your interest! 

> …and wondered if the following script alias examples are now replaceable?

The short answer is: no.

The longer answer: Even though clients are able to receive packs with something akin to `git-receive-pack`, I am pretty sure that operating it on a server requires bits and pieces of the protocol which aren't implemented yet.
And even though packs can already be created with decent performance, it's not yet able to do something equivalent to `git-upload-pack`.

Thus I am recommending to subscribe to [this tracking issue](https://github.com/Byron/gitoxide/issues/307) for updates on progress off the server side.

I hope that helps.
